corehq/apps/users/migrations/0037_add_edit_messaging_permission.py

```diff
@@ -0,0 +1,26 @@
+
+from django.db import migrations
+
+from corehq.apps.users.models_role import SQLPermission, UserRole
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+@skip_on_fresh_install
+def migrate_edit_migrations_permissions(apps, schema_editor):
+    permission, created = SQLPermission.objects.get_or_create(value='edit_messaging')
+    edit_data_permission = SQLPermission.objects.get(value='edit_data')
+    role_ids_with_edit_data = set(UserRole.objects.filter(rolepermission__permission_fk_id=edit_data_permission.id)
+                                  .values_list("id", flat=True))
+    for role in UserRole.objects.filter(id__in=role_ids_with_edit_data):
```

Comment:
nit: what's the purpose of returning just the IDs and then re-fetching the objects?
```suggestion
    roles = UserRole.objects.filter(rolepermission__permission_fk_id=edit_data_permission.id)
    for role in roles:
```

Reply:
Ah the idea was from this [PR](https://github.com/dimagi/commcare-hq/pull/30164/files) but I realize now that if the chunking is not being used it is not necessary

Reply:
ah, I guess that's a good point then too - _should_ this be using chunking?  As-is, we'll be fetching all roles with edit data permission into memory, which might be too big to manage on prod.

Reply:
added that change here: https://github.com/dimagi/commcare-hq/pull/30431/commits/0473702e275a19a9f776b8bffbbcb4a6986a9cc1